### ISSUE TITLE
cli can find Yubikey but not GUI

### ISSUE THREAD

**Issue by ItzMiracleOwO** [Yubico/yubikey-manager-qt#361]
- YubiKey Manager version: 1.2.5
- How was it installed?: `rpm-ostree install yubikey-manager-qt`
- Operating system and version: Fedora Silverblue 40
- YubiKey model and version: Yubikey 5C NFC
- Bug description summary:
cli can find Yubikey but not GUI

Steps to reproduce
---

[Please explain what you did when the bug appeared, and if and how you have been
able to reproduce it.]

1. plug in the yubikey
2. start yubikey manager

Expected result
--- 

[What did you expect to happen when you did the above?]
Yubikey show up in the GUI

Actual results
--- 

[What actually happened?]
Nothing Showed up

Other info
---

[Anything else you would like to add?]
![image](https://github.com/Yubico/yubikey-manager-qt/assets/76854136/8a2b5801-715b-493f-b8d4-47a3edfb9887)

**Comment by apollo13** [comment]
This is due to the usage of Python 3.12 which is not supported by pyotherside (yet, see https://github.com/thp/pyotherside/pull/131 etc).

As a workaround you can download https://raw.githubusercontent.com/Yubico/yubikey-manager-qt/main/ykman-gui/py/yubikey.py to `~/.local/lib/python3.12/site-packages`

**Comment by kimdre** [comment]
> This is due to the usage of Python 3.12 which is not supported by pyotherside (yet, see [thp/pyotherside#131](https://github.com/thp/pyotherside/pull/131) etc).
> 
> As a workaround you can download https://raw.githubusercontent.com/Yubico/yubikey-manager-qt/main/ykman-gui/py/yubikey.py to `~/.local/lib/python3.12/site-packages`

Thank you this also worked for me!
However I had to add the file to
`/usr/lib/python3.12/site-packages/yubikey.py`

**Comment by orjawell** [comment]
> > This is due to the usage of Python 3.12 which is not supported by pyotherside (yet, see [thp/pyotherside#131](https://github.com/thp/pyotherside/pull/131) etc).
> > As a workaround you can download https://raw.githubusercontent.com/Yubico/yubikey-manager-qt/main/ykman-gui/py/yubikey.py to `~/.local/lib/python3.12/site-packages`
> 
> Thank you this also worked for me! However I had to add the file to `/usr/lib/python3.12/site-packages/yubikey.py`

Thanks, it worked for me !

**Comment by rathann** [comment]
> This is due to the usage of Python 3.12 which is not supported by pyotherside (yet, see [thp/pyotherside#131](https://github.com/thp/pyotherside/pull/131) etc).
> 
> As a workaround you can download https://raw.githubusercontent.com/Yubico/yubikey-manager-qt/main/ykman-gui/py/yubikey.py to `~/.local/lib/python3.12/site-packages`

This doesn't fix the issue for me. `strace` shows that `yubikey.py` is found and opened, but the GUI still shows no key.

**Comment by leigh123linux** [comment]
I have bisected the offending commit


https://github.com/Yubico/yubikey-manager/commit/2bcf81b6dffc9ae6748eb251ff54a864b306e794

**Comment by apollo13** [comment]
This might be fixed with #363

**Comment by darix** [comment]
the patch in #363 has so many unrelated changes. not sure how this happened. the minimal patch is:

```
diff --git a/ykman-gui/py/yubikey.py b/ykman-gui/py/yubikey.py
index 2c91bc4..32a2b37 100644
--- a/ykman-gui/py/yubikey.py
+++ b/ykman-gui/py/yubikey.py
@@ -50,12 +50,10 @@ from ykman import __version__ as ykman_v
 if int(ykman_v.split(".")[0] ) > 4:
     from yubikit.support import get_name
     from ykman.device import list_all_devices, scan_devices
-    from ykman.otp import (
-    _PrepareUploadFailed as PrepareUploadFailed
-    , _prepare_upload_key as prepare_upload_key, generate_static_pw)
+    from ykman.otp import (generate_static_pw)
 else:
     from ykman import connect_to_device, scan_devices, get_name
-    from ykman.otp import PrepareUploadFailed, prepare_upload_key, generate_static_pw
+    from ykman.otp import generate_static_pw
 
 from fido2.ctap2 import Ctap2, ClientPin
 
@@ -400,17 +398,6 @@ class Controller(object):
         upload_url = None
 
         with self._open_device([OtpConnection]) as conn:
-            if upload:
-                try:
-                    upload_url = prepare_upload_key(
-                        key, public_id, private_id,
-                        serial=self._dev_info['serial'],
-                        user_agent='ykman-qt/' + app_version)
-                except PrepareUploadFailed as e:
-                    logger.debug('YubiCloud upload failed', exc_info=e)
-                    return failure('upload_failed',
-                                   {'upload_errors': [err.name
-                                                      for err in e.errors]})
             try:
                 session = YubiOtpSession(conn)
                 session.put_configuration(

```

**Comment by apollo13** [comment]
My changes in #363 are not unrelated, your minimal patch just fixes the python code while still leaving a broken UI if you click in the wrong buttons etc. my patch also removes the functionality from the UI

**Comment by darix** [comment]
the changes in .github/workflow/ seem unrelated still.

**Comment by apollo13** [comment]
It is not really unrelated. It removes a patch for macosx that was needed for the otp upload functionality which your and my patch removes.

**Comment by apollo13** [comment]
FWIW yubikey-manager-qt is apparently deprecated since three months: https://github.com/Yubico/yubikey-manager-qt/commit/28dc02d11b081683b59d16d12043aaa3c0a6c75f 

As such I have switched to the Yubico Authenticator which works well enough. So instead of trying to patch this in distros, they should probably get rid of this package and switch over to the authenticator.

**Comment by darix** [comment]
yeah. I already asked them to maybe archive the now deprecated projects via https://github.com/Yubico/yubikey-manager/issues/627
